File: python/src/safe_calculator.py
class Authorizer:
    def authorize(self):
        # in real life it would do some fancy lookup accross the network
        raise RuntimeError("not implemented yet")


class SafeCalculator:
    def __init__(self, authorizer: Authorizer):
        self.authorizer = authorizer


    def add(self, left, right):
        authorized = self.authorizer.authorize()
        # Bug! Should be `if not authorized`
        if authorized:
            raise RuntimeError("Not authorized")
        return left + right

File: ruby/lib/safe_calculator.rb
# frozen_string_literal: true

class Authorizer
  def authorize
    # in real life it would do some fancy lookup accross the network
    raise StandardError, 'not implemented yet'
  end
end

class SafeCalculator
  def initialize(authorizer)
    @authorizer = authorizer
  end

  def add(left, right)
    authorized = @authorizer.authorize

    # Bug! Should be `if !authorized`
    raise StandardError, 'Not authorized' if authorized

    left + right
  end
end
